### PR TITLE
Deprecate boolean-returning variants of the channel.write_ methods

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
The current channel.write_ methods return an always-true boolean. These include: 
-`write` 
-`writeln`
-`writebits` 
-`writeBytes` 
-`writef` in IO.FormattedIO 
-`writef` in ChapelIO (top level)

This PR deprecates those methods and replaces them with non-returning versions (based on this issue: https://github.com/chapel-lang/chapel/issues/19490)

Because the new signatures only differ from the old ones in return type, I've created a new parameter: `IO.WritersReturnBool` to toggle between compiling with the new and deprecated overloads of these methods using where-clauses. It currently defaults to `true` (i.e. continue to use the deprecated methods unless the user decides to compile with it set to `false`).

I've also  created some undocumented helper methods that don't have where-clauses or deprecation messages (`_write`, `_writeln`, `_writeBytes`, `_writef`). All the module code that previously called into one of these methods (including the top level write, writeln, and writef in ChapelIO), now call the relevant helper method. This way we don't throw deprecation messages from internal code regardless of the value of `WritersReturnBool`.

As another important note, I put a document explaining the details of how to complete this deprecation under `test/deprecation/IO/writersReturnBool.md`. This is a note-to-self, for whenever I remove the old returning writer methods in a later release, but it will also help to summarize this PR to whomever reviews it. 

- [x] Paratest Passing (including Mason tests - `CHPL_RE2=bundled`)

### CODE REVIEW

File: test/deprecated/IO/writersReturnBool.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+All Modules (and tools/mason/*)
+---
+
+Switch `ch._write`, `ch._writeln`, `ch._writef`, and `ch._writeBytes` calls back to `ch.write`, `ch.writeln`, `ch.writef` and `ch.writeBytes`
+
+
+Tests
+---
+
+1. Remove the COMPOPTS file with `-s WritersReturnBool=false` from the test directory
```
Comment (lydia-duncan) [inline test/deprecated/IO/writersReturnBool.md:29]: If start_test has been modified, why does `test/` need a `COMPOPTS` file that sets it?

File: test/deprecated/IO/writersReturningBool.chpl [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+var w = openwriter("./wrb.txt");
+
+w.write("a");
+w.writeln("b");
+w.writebits(2, 2);
+w.writeBytes(0xF, 1);
+w.writef("%i", 5);
+
+w.close(); remove("./wrb.txt");
```
Comment (lydia-duncan) [inline test/deprecated/IO/writersReturningBool.chpl:13]: It's unlikely to be a problem, but I think we probably should verify these still perform their operations appropriately, so would rather we check the contents of the `.txt` file than just delete it.  You can do that with a `.catfiles` for the test, or you can read it in and check you got what you expected

**Review by lydia-duncan — COMMENTED** [review]
This was a lot of work, thanks for taking it on!  I had a couple of suggestions, but otherwise I think this is in good shape!

File: test/deprecated/IO/writersReturnBool.md (hunk 1 above)
Comment (jeremiah-corrado) [reply test/deprecated/IO/writersReturnBool.md:29]: This was sort of in there for redundancy, but it can be removed.

**Comment by jeremiah-corrado** [comment]
That test has been updated, and I removed the redundant compopts file. Let me know if you see anything else.

I'll run a paratest in a couple more configs before merging since this has the potential to cause a lot of noise if it does fail.

**Review by lydia-duncan — APPROVED** [review]
Sweet, thanks!

**Pull request merged**